ingestion/src/metadata/utils/connections.py

```diff
@@ -364,6 +368,54 @@ def _(connection: MetabaseClient) -> None:
             connection.client["connection"].hostPort + "/api/dashboard",
             headers=connection.client["metabase_session"],
         )
+    except Exception as err:
+        raise SourceConnectionException(
+            f"Unknown error connecting with {connection} - {err}."
+        )
+
+
+@get_connection.register
+def _(connection: TableauConnection, verbose: bool = False):
+
+    from tableau_api_lib import TableauServerConnection
+
+    tableau_server_config = {
+        f"{connection.env}": {
+            "server": connection.hostPort,
+            "api_version": connection.apiVersion,
+            "site_name": connection.siteName,
+            "site_url": connection.siteName,
+        }
+    }
+    if connection.username and connection.password:
+        tableau_server_config[connection.env]["username"] = connection.username
+        tableau_server_config[connection.env][
+            "password"
+        ] = connection.password.get_secret_value()
+    elif connection.personalAccessTokenName and connection.personalAccessTokenSecret:
+        tableau_server_config[connection.env][
+            "personal_access_token_name"
+        ] = connection.personalAccessTokenName
+        tableau_server_config[connection.env][
+            "personal_access_token_secret"
+        ] = connection.personalAccessTokenSecret
+    try:
+        conn = TableauServerConnection(
+            config_json=tableau_server_config,
+            env=connection.env,
+        )
+        conn.sign_in().json()
+        return TableauClient(conn)
+    except Exception as err:  # pylint: disable=broad-except
+        logger.error("%s: %s", repr(err), err)
+
+
+@test_connection.register
+def _(connection: TableauClient) -> None:
+    from tableau_api_lib.utils.querying import get_workbooks_dataframe
+
+    try:
+        get_workbooks_dataframe(connection.client).to_dict()
```

Comment:
a quick comment please, if I may. Do you know how much load this `get_workbooks_dataframe(connection.client).to_dict()` requires? Without knowing how the tableau API works, it seems like this function might take a lot of time if we have a lot of workbooks. Also pandas is not always super fast.

Is there any other function we might use to just check for the server version or something that does not move much data?

Thanks